runelite-client/src/main/java/net/runelite/client/plugins/hiscore/HiscorePlugin.java

```diff
@@ -219,25 +226,47 @@ public void onLookupMenuClicked(PlayerMenuOptionClicked event)
 	{
 		if (event.getMenuOption().equals(LOOKUP))
 		{
-			executor.execute(() ->
+			lookupPlayer(Text.removeTags(event.getMenuTarget()));
+		}
+	}
+
+	private void lookupPlayer(String playerName)
+	{
+		executor.execute(() ->
+		{
+			try
 			{
-				try
+				SwingUtilities.invokeAndWait(() ->
 				{
-					SwingUtilities.invokeAndWait(() ->
+					if (!navButton.isSelected())
 					{
-						if (!navButton.isSelected())
-						{
-							navButton.getOnSelect().run();
-						}
-					});
-				}
-				catch (InterruptedException | InvocationTargetException e)
+						navButton.getOnSelect().run();
+					}
+				});
+			}
+			catch (InterruptedException | InvocationTargetException e)
+			{
+				throw new RuntimeException(e);
+			}
+
+			hiscorePanel.lookup(playerName);
+		});
+	}
+
+	@Subscribe
+	public void onChatMessageReceived(ChatMessage event)
+	{
+		if (config.bountylookup() && event.getType().equals(ChatMessageType.SERVER))
```

Comment:
Reverse these conditions--check for message type before parsing the text.

Reply:
Wouldn't it be more efficient to first check if messages need to be parsed at all before checking what kind of chatmessage it is?

Reply:
Doing a regex parse is almost certainly more expensive than an equality check.

Reply:
I'm not sure I understand how reversing the conditions would influence that. I have pushed an updated version of the code, does that fix the problem?

Reply:
[Because of short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation), if the first conditional is false, it would not evaluate the second condition, hence the cheaper and/or more-likely-to-fail operation should be evaluated first.

Reply:
That said, I did misread the original code--looks like it's ordered correctly, and the code looks fine now. I had thought for some reason that the first check was a regex matcher instead of the config check. My bad there. :upside_down_face: